lib/middlewares/src/metering.rs

```diff
@@ -418,21 +411,21 @@ mod tests {
             .unwrap();
         add_one.call(&mut ctx.as_context_mut(), 1).unwrap();
         assert_eq!(
-            get_remaining_points(ctx.as_context_mut(), &instance),
+            get_remaining_points(&mut ctx.as_context_mut(), &instance),
             MeteringPoints::Remaining(6)
         );
 
         // Second call
         add_one.call(&mut ctx.as_context_mut(), 1).unwrap();
         assert_eq!(
-            get_remaining_points(ctx.as_context_mut(), &instance),
+            get_remaining_points(&mut ctx.as_context_mut(), &instance),
             MeteringPoints::Remaining(2)
         );
 
         // Third call fails due to limit
-        assert!(add_one.call(&mut ctx.as_context_mut(), 1).is_err());
+        assert!(add_one.call(1).is_err());
```

Comment:
Is this call valid?

Reply:
No. That's strange. Something must have gone wrong when I  rebased the branch over context-api.